telemetry_package/main.py

```diff
@@ -1,28 +1,73 @@
-import sys
+import json
 import rclpy
-from rclpy.node import Node
+import socket
+import os
+import select
 
+from rcl_interfaces.msg import Log
 from common_package_py.common_node import CommonNode
 
+DEFAULT_UNIX_SOCKET_PATH = "/tmp/thi_drone"
 
 class TelemetryNode(CommonNode):
-    """A class that represents a telemetry node.
-    """
 
-    def __init__(self, id: str):
-        """Creates a new TelemetryNode.
+    def __init__(self):
+        # The creation of the needed sockets takes place in the constructor
+        super().__init__('telemetry_node')
 
-        Args:
-            id (str): Unique node id
-        """
+        if os.path.exists(DEFAULT_UNIX_SOCKET_PATH): 
+            raise FileExistsError('Socket already exists')
 
-        super().__init__(id)
+        try:         
+            # Create server socket
+            self.server_sock = socket.socket(socket.AF_UNIX, socket.        SOCK_STREAM)
+
+            # Bind server socket and listen for a connection. The node is blocked as long as a connection does not exist
+            self.server_sock.bind(DEFAULT_UNIX_SOCKET_PATH)
+
+            print("Blocking until connection is established...")
+            self.server_sock.listen(1)
+
+            # Getting the reference for the client socket
+            readable, _, _, = select.select([self.server_sock], [], [])
+            if self.server_sock in readable:            
+                # Accept the connection
+                self.client_sock, _ = self.server_sock.accept()
+                print('Connection accepted. Creating Subscription')
+
+                
+            # Creating the subscription after the connection has accepted ensures that the callback function is executed only after the socket is connected           
+            self.subscription = self.create_subscription(
+                Log,
+                '/rosout',
+                self.subscription_callback, 
+                100)
+            self.subscription
```

Comment:
Here we need to add a subscription to the heartbeat topic, to satisfy Philipp wish to see status of nodes

+ Add QoS Settings here